src/main/java/com/greenfox/poker/service/GameService.java

```diff
@@ -1,82 +1,150 @@
+    return (gameRepo.exists(id));
+  }
+
+  public boolean isGameExistByName(String name) {
+    if (gameRepo.findAllByName(name).size() > 0) {
       return true;
     }
     return false;
   }
 
-  public Game getGameById(long id) {
-    return gameRepo.findOne(id);
+  public Game getGameById(long gameId) {
+    return gameRepo.findOne(gameId);
   }
 
-  public GameState getGameState(long id) {
-    return gameStateHashMap.get(id);
+  public GameState getGameStateById(long id) {
+    return gameStateMap.get(id);
   }
 
-  public ResponseEntity<?> getGameStateById(long id) {
-    if (isGameExist(id)) {
-      long currentStateId = getGameById(id).getGamestateId();
-      return new ResponseEntity(getGameState(currentStateId), HttpStatus.OK);
+  public boolean isPlayerInTheGame(long gameId, long userId) {
+    if (!gameStateMap.isEmpty()) {
+      if (!gameStateMap.get(gameId).getPlayers().isEmpty()) {
+        for (GamePlayer player : gameStateMap.get(gameId).getPlayers()) {
+          if (player.getId() == userId) {
+            return true;
+          }
+        }
+      }
     }
-    return new ResponseEntity(new StatusError("fail", "game id doesn’t exist"),
-            HttpStatus.NOT_FOUND);
+    return false;
+  }
+
+  public void createNewPlayerAndAddToGame(long playerId, long gameId, long chipsToPlayWith){
+    PokerUserDTO player = dtoService.userDTOHashMap.get(playerId);
+    GamePlayer newPlayer = new GamePlayer(chipsToPlayWith, player);
+    gameStateMap.get(gameId).getPlayers().add(newPlayer);
+  }
+
+  public ResponseType joinPlayerToGame(long playerId, long gameId, long chipsToPlayWith){
+    String gameName = gameRepo.findOne(gameId).getName();
+    String playerName = dtoService.userDTOHashMap.get(playerId).getUsername();
+    dtoService.deductChipsFromAvailableChips(chipsToPlayWith, playerId);
+    createNewPlayerAndAddToGame(playerId,gameId,chipsToPlayWith);
+    return new StatusError("success", playerName + " joined game: " + gameName);
+  }
+
+  public int getTableBigBlind (long gameId){
+    return gameRepo.findOne(gameId).getBigBlind();
   }
 
-  public ResponseEntity<?> saveNewGame(Game game, BindingResult bindingResult) {
-    if (bindingResult.hasErrors()) {
-      return new ResponseEntity(errorMessageService.respondToMissingParameters(bindingResult),
-              HttpStatus.BAD_REQUEST);
-    } else {
-      return new ResponseEntity(saveGame(game), HttpStatus.OK);
+  public ResponseType leaveGame(long gameId, long playerId){
+    List<GamePlayer> playersAtTable = getGameStateById(gameId).getPlayers();
+    for (GamePlayer player : playersAtTable){
+      if (playerId == player.getId()){
+        playersAtTable.remove(player);
+      }
     }
+    getGameStateById(gameId).setPlayers(playersAtTable);
+    String playerName = dtoService.userDTOHashMap.get(playerId).getUsername();
+    String gameName = gameRepo.findOne(gameId).getName();
+    return new StatusError("success", playerName + " left game: " + gameName);
```

Comment:
You should separate the "leave game" logic from the response you send after a successful leaving.